### DISCUSSION TITLE
How to access "instance" in a class method?

### DISCUSSION THREAD

**Discussion by pauloborges** [apple/pkl#320] [Q&A] [answered]
Hello folks, I tried to find out if this has been asked before, but I couldn't find anything.

How do I access the whole class instance using a method? For example:

```pkl
class Foo {
  bar: String

  function hello(): String = fooer(???)
}

function fooer(foo: Foo): String = throw("not implemented")
```

**Comment by bioball** [comment, marked as answer]
`fooer(this)` would pass in this instance of `Foo`.

Also, in order for this to work, you'll need to make `fooer` `const`:

```diff
-function fooer(foo: Foo): String = throw("not implemented")
+const function fooer(foo: Foo): String = throw("not implemented")
```

**Reply by pauloborges** [reply]
Thanks!
